### PR TITLE
Add unit tests for repositories

### PR DESCRIPTION
**Pull request by STALINovsky**
Add class to assist in testing repositories & implement TaskStateRepositoryTest, TaskTrackRepositoryTest.

### CODE REVIEW

**Review by ArtyomKolosov2 — CHANGES_REQUESTED** [review]
Hi. I did some comments on your PR. Some of your namings in tests aren't obvious as for me. Please, improve them.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Infrastructure/RepositoryTestHelper.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            await _context.SaveChangesAsync();
+
+            //Act
+            var repositoryEntity = await DbSet.FirstAsync();
+
+            //Assert
+            Assert.Equal(entity, repositoryEntity, _comparer);
+        }
+
+        public async Task IsEntityUpdated(TEntity startEntityState, Func<TEntity, TEntity> changeState)
```
Comment (ArtyomKolosov2) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Infrastructure/RepositoryTestHelper.cs:80]: Why this method called "IsEntityUpdated", but it doesn't returns boolean value?

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Infrastructure/RepositoryTestHelper.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        /// <param name="repository"></param>
+        /// <param name="deconstruct">lambda expressions which deconstruct entity to ValueTuple</param>
+        public RepositoryTestHelper(DIMSCoreContext context, IRepository<TEntity> repository, Func<TEntity, ITuple> deconstruct)
+        {
+            _context = context;
+            _repository = repository;
+            _comparer = new EqualityComparator<TEntity>(deconstruct);
+        }
+
+        public async Task GetAllEqualsSeedData(ICollection<TEntity> seedData)
```
Comment (ArtyomKolosov2) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Infrastructure/RepositoryTestHelper.cs:41]: This naming looks quite strange

**Comment by PushMeNow** [comment]
@STALINovsky I'm closing this PR because there are too many marks which need to fix. You are using incorrect approach for testing. Contact with @dbogdan2000 he will explain what is incorrect and how to write correct test.

**Pull request closed without merging**